src/ValidationError.php

```diff
@@ -2,23 +2,32 @@
 
 namespace Schemer;
 
-class ValidationError
+final class ValidationError
 {
+    const NOT_ALPHANUMERIC = 'NOT_ALPHANUMERIC';
+    const NOT_EMAIL = 'NOT_EMAIL';
+    const LENGTH_MISMATCH = 'LENGTH_MISMATCH';
+    const NOT_LOWERCASE = 'NOT_LOWERCASE';
+    const NOT_UPPERCASE = 'NOT_UPPERCASE';
+    const TOO_LONG = 'TOO_LONG';
+    const TOO_SHORT = 'TOO_SHORT';
+    const REGEX_MISMATCH = 'REGEX_MISMATCH';
```

Comment:
The solution that comes to mind is to define your set of translatable tokens in each validator, and then the result object looks in a given translation set for the translations it needs. That way, you could limit the bloat, and return the consts to literal strings (which I know you hate BUT it would mean that the token itself is always the fallback if no translation be found).